examples/documentation/require_started_documentation.cpp

```diff
@@ -0,0 +1,54 @@
+//  Copyright (c) 2024 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <pika/execution.hpp>
+#include <pika/init.hpp>
+
+#include <fmt/printf.h>
+#include <fmt/ranges.h>
+
+#include <cassert>
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+
+int main(int argc, char* argv[])
+{
+    namespace ex = pika::execution::experimental;
+    namespace tt = pika::this_thread::experimental;
+
+    pika::start(argc, argv);
+    ex::thread_pool_scheduler sched{};
+
+    {
+        // require_started forwards values received from the predecessor sender
+        auto rs1 = ex::just(42) | ex::require_started() |
+            ex::then([]([[maybe_unused]] auto&& i) { assert(i == 42); });
+        tt::sync_wait(std::move(rs1));
+    }
+
+    {
+        // The termination is ignored with discard, the sender is from the user's perspective
+        // rightfully not started/connected
+        auto rs2 = ex::just() | ex::require_started();
+        rs2.discard();
+    }
+
+    {
+        // require_started terminates on destruction if not connected/started
+        std::set_terminate([] {
+            std::cout << "std::terminate called: rs3 never connected or started\n";
+            pika::finalize();
+            pika::stop();
+            std::exit(1);
+        });
+
+        auto rs3 = ex::just() | ex::require_started();
+    }
+    assert(false);
+
+    return 0;
+}
```

Comment:
Yes and no. I would actually put them at the end of main (i.e. after `assert(false)` and before `return 0`) where they belong under normal circumstances (I missed this). But I would remove them from the terminate call since at that point we just want to exit the program.